cmd/src/debug_common.go

```diff
@@ -0,0 +1,470 @@
+	getPods.Stdout = &podsBuff
+	getPods.Stderr = os.Stderr
+	err := getPods.Run()
+
+	//Declare struct to format decode from podList
+	var pods podList
+
+	//Decode json from podList
+	if err := json.NewDecoder(&podsBuff).Decode(&pods); err != nil {
+		fmt.Errorf("failed to unmarshall get pods json: %w", err)
+	}
+
+	return pods, err
+}
+
+func getEvents(ctx context.Context, namespace, baseDir string) *archiveFile {
+	f := &archiveFile{name: baseDir + "/kubectl/events.txt"}
+	f.data, f.err = exec.CommandContext(ctx, "kubectl", "-n", namespace, "get", "events").CombinedOutput()
+	return f
+}
+
+func getPV(ctx context.Context, namespace, baseDir string) *archiveFile {
+	f := &archiveFile{name: baseDir + "/kubectl/persistent-volumes.txt"}
+	f.data, f.err = exec.CommandContext(ctx, "kubectl", "-n", namespace, "get", "pv").CombinedOutput()
+	return f
+}
+
+func getPVC(ctx context.Context, namespace, baseDir string) *archiveFile {
+	f := &archiveFile{name: baseDir + "/kubectl/persistent-volume-claims.txt"}
+	f.data, f.err = exec.CommandContext(ctx, "kubectl", "-n", namespace, "get", "pvc").CombinedOutput()
+	return f
+}
+
+// get kubectl logs for pod containers
+func getContainerLog(ctx context.Context, podName, containerName, namespace, baseDir string) *archiveFile {
+	f := &archiveFile{name: baseDir + "/kubectl/pods/" + podName + "/" + containerName + ".log"}
+	f.data, f.err = exec.CommandContext(ctx, "kubectl", "-n", namespace, "logs", podName, "-c", containerName).CombinedOutput()
+	return f
+}
+
+// get kubectl logs for past container
+func getPastContainerLog(ctx context.Context, podName, containerName, namespace, baseDir string) *archiveFile {
+	f := &archiveFile{name: baseDir + "/kubectl/pods/" + podName + "/" + "prev-" + containerName + ".log"}
+	cmdStr := []string{"kubectl", "-n", namespace, "logs", "--previous", podName, "-c", containerName}
+	f.data, f.err = exec.CommandContext(ctx, "kubectl", "-n", namespace, "logs", "--previous", podName, "-c", containerName).CombinedOutput()
+	if f.err != nil {
+		f.err = errors.Wrapf(f.err, "executing command: %s: received error: %s", cmdStr, f.data)
+	}
+	return f
+}
+
+func getDescribe(ctx context.Context, podName, namespace, baseDir string) *archiveFile {
+	f := &archiveFile{name: baseDir + "/kubectl/pods/" + podName + "/describe-" + podName + ".txt"}
+	f.data, f.err = exec.CommandContext(ctx, "kubectl", "-n", namespace, "describe", "pod", podName).CombinedOutput()
+	return f
+}
+
+func getManifest(ctx context.Context, podName, namespace, baseDir string) *archiveFile {
+	f := &archiveFile{name: baseDir + "/kubectl/pods/" + podName + "/manifest-" + podName + ".yaml"}
+	f.data, f.err = exec.CommandContext(ctx, "kubectl", "-n", namespace, "get", "pod", podName, "-o", "yaml").CombinedOutput()
+	return f
+}
+
+/*
+Docker functions
+TODO: handle for single container/server instance
+*/
+
+func archiveDocker(ctx context.Context, zw *zip.Writer, verbose, configs bool, baseDir string) error {
```

Comment:
The general structure of all the `archiveX` functions are the same. Have you thought about making an abstraction that does the common "writes to zip file from channel contents" suffix?

Reply:
done